### DISCUSSION TITLE
Filter IP addresses by parent prefix in custom scripts

### DISCUSSION THREAD

**Discussion by JulijusBertasius** [netbox-community/netbox#9487] [Q&A] [answered]
I can't find a easy way to get all IP addresses by parent prefix in custom scripts.

There is no option like:
````
IPAddress.objects.filter(prefix=parent_prefix)
````

**Comment by candlerb** [comment, marked as answer]
There's a [method](https://github.com/netbox-community/netbox/blob/v3.2.4/netbox/ipam/models/ip.py#L486-L494) the other way round, on Prefix:

```
addrs = parent_prefix.get_child_ips()
```

If you still want to start with `IPAddress.objects` and filter downwards, then I suggest you copy the logic from the REST API `IPAddressFilterSet` [here](https://github.com/netbox-community/netbox/blob/v3.2.4/netbox/ipam/filtersets.py#L467-L470) and [here](https://github.com/netbox-community/netbox/blob/v3.2.4/netbox/ipam/filtersets.py#L573-L581). The key part is:

```
            query = str(netaddr.IPNetwork(value.strip()).cidr)
            return queryset.filter(address__net_host_contained=query)
```

(you may want to filter on the value of VRF as well)

**Reply by Domoninic** [reply]
@JulijusBertasius :  I am looking for this same functionality in a script. Were you able to get this to work ?

**Reply by JulijusBertasius** [reply]
Yes, it works for me.

`all_ip = IPAddress.objects.filter(address__net_host_contained=str(parent_prefix))`
